middlewares/authenticateTwitter.js

```diff
@@ -0,0 +1,49 @@
+import request from 'request';
+
+export const sendTwitterUser = (req, res, next) => {
+  request.post({
+    url: 'https://api.twitter.com/oauth/request_token',
+    oauth: {
+      oauth_callback: 'http%3A%2F%2Flocalhost%3A8080%2Ftwitter-callback',
```

Comment:
What is this `oauth_callback` about? Must the url look that random?

Reply:
The url is encoded. It would resolve to `http://localhost:8080/twitter-callback`.

@XROLE This should also not be hardcoded so that we can easily update it per environment.

Reply:
Okay @andela-mabdussalam. Got it.

Reply:
`react-twitter-auth`nwhich is the package to be consumed in the frontend will check if URL is still on api.twitter.com, and if it is not it will take parameters from URL.
I got to know this from the comments of this [Article](https://medium.com/@robince885/how-to-do-twitter-authentication-with-react-and-restful-api-e525f30c62bb)

Reply:
Also, it looks to me like you hardcoded `8080` in that encoded url. Can you store the port in `.env` instead?

Reply:
Ok.

Reply:
The `oauth_callback`? I cannot see the change.

Reply:
Oh, not the `oauth_callback`. I will change that now.

Reply:
I just change it